tests/sqlite/sqlite.rs

```diff
@@ -960,3 +960,36 @@ async fn test_multiple_set_rollback_hook_calls_drop_old_handler() -> anyhow::Res
     assert_eq!(1, Arc::strong_count(&ref_counted_object));
     Ok(())
 }
+
+#[sqlx_macros::test]
+async fn it_can_use_transaction_options() -> anyhow::Result<()> {
+    use sqlx_sqlite::SqliteTransactionState;
+
+    async fn check_txn_state(
+        conn: &mut SqliteConnection,
+        expected: SqliteTransactionState,
+    ) -> Result<(), sqlx::Error> {
+        let state = conn.lock_handle().await?.transaction_state()?;
+        assert_eq!(state, expected);
+        Ok(())
+    }
+
+    let mut conn = new::<Sqlite>().await?;
+
+    check_txn_state(&mut conn, SqliteTransactionState::None).await?;
+
+    let mut tx = conn.begin_with("BEGIN DEFERRED").await?;
+    check_txn_state(&mut *tx, SqliteTransactionState::None).await?;
+    drop(tx);
+
+    let mut tx = conn.begin_with("BEGIN IMMEDIATE").await?;
+    check_txn_state(&mut *tx, SqliteTransactionState::Write).await?;
+    drop(tx);
+
+    // Note: may result in database locked errors if tests are run in parallel
```

Comment:
The nice thing about SQLite is we can just use an in-memory database, or a temporary one per-test. You don't _need_ to use `new::<Sqlite>()` here, especially since it doesn't really depend on any existing schema.